jupyter-extension/jupyterlab_pachyderm/tests/test_config.py

```diff
@@ -0,0 +1,69 @@
+"""Tests for the config screen and handler."""
+from pathlib import Path
+
+import pytest
+from httpx import AsyncClient
+from tornado.web import Application
+
+from pachyderm_sdk.config import ConfigFile
+
+from jupyterlab_pachyderm.env import PACH_CONFIG
+
+
+@pytest.mark.no_config
+async def test_config_no_file(app: Application, http_client: AsyncClient):
+    """Test that if there is no config file present, the extension does not
+    use a default config."""
+    # Arrange
+    config_file = app.settings.get("pachyderm_config_file")
+    assert config_file is not None and not config_file.exists()
+
+    # Act
+    response = await http_client.get("/config")
+
+    # Assert
+    response.raise_for_status()
+    payload = response.json()
+    assert payload["cluster_status"] == "INVALID"
+    assert payload["pachd_address"] == ""
+
+
+@pytest.mark.no_config
+async def test_do_not_set_invalid_config(app: Application, http_client: AsyncClient):
+    """Test that PUT /config does not store an invalid config."""
+    # Arrange
+    assert app.settings.get("pachyderm_client") is None
+    invalid_address = "localhost:33333"
+    data = {"pachd_address": invalid_address}
+
+    # Act
+    response = await http_client.put("/config", json=data)
+
+    # Assert
+    response.raise_for_status()
+    payload = response.json()
+    assert payload["cluster_status"] == "INVALID"
+    assert payload["pachd_address"] == invalid_address
+    assert app.settings.get("pachyderm_client") is None
```

Comment:
can we also check that the config file was not written to?

Reply:
great point, I'll add this